.github/workflows/neon-branch-cleanup.yml

```
@@ -0,0 +1,23 @@
+      - name: Delete Neon Branch
+        id: delete_neon_branch
+        continue-on-error: true
+        uses: neondatabase/delete-branch-action@v3
+        with:
+          project_id: ${{ secrets.NEON_PROJECT_ID }}
+          api_key: ${{ secrets.NEON_API_KEY }}
+          branch: preview/${{ github.event.pull_request.head.ref }}
+      
+      - name: Warn if Neon branch was not deleted
+        if: steps.delete_neon_branch.outcome == 'failure'
+        run: |
```

actionlint (1.7.9)

"github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)